### ISSUE TITLE
[Question] How to simply get the "Union"?

### ISSUE THREAD

**Issue by chaojian-zhang** [cutdigital/mcut#8]
Hi, thanks again for the great library and sorry for asking such a basic question😂

According to the **Results** section of the [tutorial](https://cutdigital.github.io/mcut.site/tutorials/helloWorld/#results) - there are a wide variety of potential outputs (depending on the boolean method): however, I do not quite see how we can *specify which of those outputs we want to get*.

I suppose those are already all returned inside the `connComps` when we query it like below:

```c++
uint32_t numConnComps;
std::vector<McConnectedComponent> connComps;
mcGetConnectedComponents(context, MC_CONNECTED_COMPONENT_TYPE_ALL, 0, NULL, &numConnComps);
connComps.resize(numConnComps);
mcGetConnectedComponents(context, MC_CONNECTED_COMPONENT_TYPE_ALL, (uint32_t)connComps.size(), connComps.data(), NULL);
```

However, I am having difficulty differentiating which component is which. I saw some enums that seems to help differentiating between various components but I am not sure how to use them (also I am not sure whether it's implementation detail or query parameters).

```c++
typedef enum McFragmentLocation {
    MC_FRAGMENT_LOCATION_ABOVE = 1 << 0, /**< Fragment is located above the cut-mesh. */
    MC_FRAGMENT_LOCATION_BELOW = 1 << 1, /**< Fragment is located below the cut-mesh. */
    MC_FRAGMENT_LOCATION_UNDEFINED = 1 << 2, /**< Fragment is located neither above nor below the cut-mesh. That is, it was produced due to a partial cut intersection. */
    MC_FRAGMENT_LOCATION_ALL = 0xFFFFFFFF /**< Wildcard (match all) . */
} McFragmentLocation;
``` 

In short, the question is, **given any connComps, e.g. `connComps[0]` - how do I know which boolean operation is it resulted from**?

Also, I don't know why😅, I can't seem to find the definition for `McConnectedComponent`, all I can find is [below](https://github.com/cutdigital/mcut/blob/c3617d8c77bc3b40c9a77d2f91dfe1b3249ce8d4/include/mcut/mcut.h#L82) - is this some kind of C++ magic🥲:

```c++
typedef struct McConnectedComponent_T* McConnectedComponent;
```

Thanks!

**Comment by chitalu** [comment]
Hi,

The CSGBoolean [tutorial](https://github.com/cutdigital/mcut/tree/master/tutorials/CSGBoolean) provides an example reference on how to perform boolean operations with MCUT. 

Regarding the flags, surface normals (as determined from the polygon winding order) are important for determining what is "above"/"below" and/or "inside"/"outside".

here is an example of what I mean:

![image](https://user-images.githubusercontent.com/7461679/188766641-217244d1-cc03-432e-be97-3a60b0e30afc.png)

In this case, I have drawn the surface normals as little arrows on each mesh for your reference. The terms "above" and "below" apply only to fragments, where a fragment is above if it is on the side of the cut-mesh that is pointed-to by the normals of this cut-mesh. Likewise it is below if on the side that is opposite to the direction of the normals. It is important to bare in mind that this information is detected/inferred from the polygons/faces that are intersecting and then propagation to the rest that are on the same side of the cut-mesh (i.e. treating the intersection contour/edges as a boundary).

Similarly the notion of "inside" or "outside" applies only to patches, where a patch is outside if it is on the side pointed to by the normals of the source-mesh. Likewise it is inside if on the side that is opposite to the direction of the normals.

I hope that from this is easy enough to infer what the notion of sealing a fragment from "inside"/"outside" means, and how this concept generalizes to mesh booleans. To further clarify this, the following is another image showing how the concepts generalize to solids and boolean operations. 

![cut-with-closed-surface](https://user-images.githubusercontent.com/7461679/188766812-dcac358c-4809-4d1e-9ea7-27e70323a687.png)

I hope this information is helpful.

best.

**Comment by chitalu** [comment]
...and here is a slightly more complex case:

![cut-with-closed-through-surface](https://user-images.githubusercontent.com/7461679/189607641-7bc7b0bd-ba20-4043-90fc-dfe884286d0a.png)
